### ISSUE TITLE
v0.3 FTBFS on ARM, mismatched signness of char*

### ISSUE THREAD

**Issue by OPNA2608** [jturcotte/chiptrack#5]
This happens in afew places, here's a full log: https://logs.ofborg.org/?key=nixos/nixpkgs.320790&attempt_id=08cf329f-7918-47c7-ae5e-9b1d87807a9e

```
error[E0308]: mismatched types
  --> src/synth_script/wasm_host.rs:43:22
   |
43 |     f(CStr::from_ptr(v1));
   |       -------------- ^^ expected `*const u8`, found `*const i8`
   |       |
   |       arguments to this function are incorrect
   |
   = note: expected raw pointer `*const u8`
              found raw pointer `*const i8`
note: associated function defined here
  --> /build/rustc-1.78.0-src/library/core/src/ffi/c_str.rs:274:25
```

AFAIU, this is about the signness of `char` in C being implementation-defined, with ARM compilers usually using `unsigned` while x86 ones use `signed`.

**Comment by jturcotte** [comment]
Seems like I should be using `core::ffi::c_char` instead of i8 in those places, I don't remember having a reason not to use it.

**Comment by jturcotte** [comment]
I could reproduce the issue on a raspberry pi. I pushed a [v0.3.1](https://github.com/jturcotte/chiptrack/releases/tag/v0.3.1) tag with the fix on top of v0.3.
